test/unit/lib/plugins/aws/invoke.test.js

```diff
@@ -73,172 +74,11 @@ describe('AwsInvoke', () => {
-    });
-
-    it('should not throw error when there is no input data', async () => {
-      awsInvoke.options.data = undefined;
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.equal('');
-    });
-
-    it('should keep data if it is a simple string', async () => {
-      awsInvoke.options.data = 'simple-string';
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.equal('simple-string');
-    });
-
-    it('should parse data if it is a json string', async () => {
-      awsInvoke.options.data = '{"key": "value"}';
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.deep.equal({ key: 'value' });
-    });
-
-    it('should skip parsing data if "raw" requested', async () => {
-      awsInvoke.options.data = '{"key": "value"}';
-      awsInvoke.options.raw = true;
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.deep.equal('{"key": "value"}');
-    });
-
-    it('it should parse file if relative file path is provided', async () => {
-      serverless.serviceDir = getTmpDirPath();
-      const data = {
-        testProp: 'testValue',
-      };
-      serverless.utils.writeFileSync(
-        path.join(serverless.serviceDir, 'data.json'),
-        JSON.stringify(data)
-      );
-      awsInvoke.options.path = 'data.json';
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.deep.equal(data);
-    });
-
-    it('it should parse file if absolute file path is provided', async () => {
-      serverless.serviceDir = getTmpDirPath();
-      const data = {
-        testProp: 'testValue',
-      };
-      const dataFile = path.join(serverless.serviceDir, 'data.json');
-      serverless.utils.writeFileSync(dataFile, JSON.stringify(data));
-      awsInvoke.options.path = dataFile;
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.deep.equal(data);
-    });
-
-    it('it should parse a yaml file if file path is provided', async () => {
-      serverless.serviceDir = getTmpDirPath();
-      const yamlContent = 'testProp: testValue';
-
-      serverless.utils.writeFileSync(path.join(serverless.serviceDir, 'data.yml'), yamlContent);
-      awsInvoke.options.path = 'data.yml';
-
-      await expect(awsInvoke.extendedValidate()).to.be.fulfilled;
-
-      expect(awsInvoke.options.data).to.deep.equal({
-        testProp: 'testValue',
-      });
-    });
-
-    it('it should throw error if file path does not exist', () => {
-      serverless.serviceDir = getTmpDirPath();
-      awsInvoke.options.path = 'some/path';
-
-      return expect(awsInvoke.extendedValidate()).to.be.rejectedWith(
-        'The file you provided does not exist.'
-      );
-    });
-
     it('should resolve if path is not given', () => {
```

Comment:
Is there a reason for leaving this test here? I believe it's the only one written the old way and if we port it (or remove it if it's not really needed) we will be able to finalize refactoring of `invoke` tests (and removing whole `AwsInvoke` `describe` section)

Reply:
the reason I left it was because there was no test set up to replace it. I've refactored the remaining test and deleted the old `AwsInvoke` section 👍